src/server/resources/FoodFeedbackSentimentAnalysis.java

```diff
@@ -0,0 +1,112 @@
+package server.resources;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Map;
+import java.util.LinkedHashMap;
+
+public class FoodFeedbackSentimentAnalysis {
+
+    private static final String SENTIMENT_KEYWORDS_FILE = "D:\\Learn_and_Code\\final-assignment-june\\Cafeteria-Recommendation-System\\src\\server\\resources\\SentimentKeyWords.txt";
+
+    private Map<String, Integer> sentimentWords;
+
+    public FoodFeedbackSentimentAnalysis() {
+        this.sentimentWords = new LinkedHashMap<>();
+        loadWords();
+    }
+
+    private void loadWords() {
+        try (BufferedReader br = new BufferedReader(new FileReader(SENTIMENT_KEYWORDS_FILE))) {
+            String line;
+            System.out.println("Loading sentiment words from file: " + SENTIMENT_KEYWORDS_FILE);
+            
+            while ((line = br.readLine()) != null) {
+                String[] parts = line.trim().toLowerCase().split("\\s+");
+                if (parts.length == 2) {
+                    String word = parts[0];
+                    int weight = Integer.parseInt(parts[1]);
+                    sentimentWords.put(word, weight);
+                }
+            }
+        } catch (IOException e) {
+            System.err.println("Error reading file: " + SENTIMENT_KEYWORDS_FILE);
+            e.printStackTrace();
+        }
+    }
+
+    public String analyzeSentiment(String comment) {
```

Comment:
Check for all large functions and break down into smaller ones. Make sure the principles are being followed.